jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererUtil.kt

```diff
@@ -94,18 +99,39 @@ object CodeWhispererUtil {
                 }
             } catch (e: SsoOidcException) {
                 runInEdt {
-                    SsoConnectionExpiredDialog(project, connection).show()
+                    notifyConnectionExpired(project, connection)
                     callback()
                 }
             }
         } else if (state == BearerTokenAuthState.NOT_AUTHENTICATED) {
             runInEdt {
-                SsoConnectionExpiredDialog(project, connection).show()
+                notifyConnectionExpired(project, connection)
                 callback()
             }
         }
     }
 
+    private fun notifyConnectionExpired(project: Project, connection: ToolkitConnection?) {
+        connection ?: return
+        logoutFromSsoConnection(project, connection) {
+            UiTelemetry.click(project, "signout_codewhisperer_expired_connection")
+        }
+        notifyError(
+            message("toolkit.sso_expire.dialog.title", connection.label),
+            message("toolkit.sso_expire.dialog_message"),
+            project,
+            listOf(
+                NotificationAction.create(message("toolkit.sso_expire.dialog.yes_button")) { _, notification ->
+                    runInEdt { ToolkitAddConnectionDialog(project, connection).show() }
```

Comment:
Does this open a dialog for adding a new connection? Would users be confused?

Reply:
this should happen when 
- we fail to refresh the connection at first place
- when user click on the `authenticate` button in the expiry notification, we logout immediately from the stale connection and will fill the metadata(sso url) if it's a sso connection

Reply:
does this make sense to you Jaden

Reply:
can refer to second screen record 0:54

Reply:
I try to avoid the state where users connected with a expired connection, so i log out of them once I detected the connection unrefreshable

Reply:
I think it's ok to jump the user straight into the browser instead of the add connection dialog.

Reply:
make sense, let me update this abit

Reply:
Jaden I updated the screen recording #2 as well where I modified both `expiredAt` `refrehToken` so that the credential is ir-refreshable